core/src/main/scala/org/bitcoins/core/api/wallet/db/OutgoingTransactionDb.scala

```diff
@@ -39,14 +39,16 @@ object OutgoingTransactionDb {
       s"sentAmount ($sentAmount) cannot be greater than the amount the wallet input ($inputAmount)")
 
     val feePaid = inputAmount - totalOutput
-    val feeRate = feePaid.satoshis.toLong / tx.baseSize.toDouble
+
+    // Calculate fee rate in satoshis per byte
+    val feeRate = feePaid.satoshis.toLong / tx.byteSize.toDouble
```

Comment:
do we really not have some sort of centralized place to do this? It scares me when we just have one off fee rate calculations.

Reply:
Agreed, I think it is out of scope for this PR however, but will file an issue to do so.

Reply:
#1989 #1990